language-support/java/bindings-rxjava/src/test/scala/com/daml/ledger/rxjava/components/BotTest.scala

```diff
@@ -43,11 +44,27 @@ import scala.concurrent.{Future, Promise}
 import scala.util.Random
 import scala.util.control.NonFatal
 
-final class BotTest extends FlatSpec with Matchers with Eventually {
+final class BotTest extends FlatSpec with Matchers with Eventually with BeforeAndAfterEach {
   override implicit def patienceConfig: PatienceConfig = PatienceConfig(5.seconds, 1.second)
 
   private def ledgerServices: LedgerServices = new LedgerServices("bot-test")
 
+  private var dispose: Seq[Disposable] = _
+
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+    this.dispose = Vector.empty
+  }
+
+  override def afterEach(): Unit = {
+    dispose.foreach(_.dispose())
+    super.afterEach()
+  }
```

Comment:
This looks odd. I'd consider either a `scala.collection.mutable.Seq` or a `Seq.newBuilder`.

Although I think using a loan pattern here would be much cleaner. E.g.:

```scala
using(Bot.wire(...)) { connection =>
}
```

Reply:
Yeah, that's much nicer. I'll do that.

Reply:
Done.